tests/test_magic_date_type.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+from datetime import timedelta
+
+from .conftest import begin_date_str
+from .conftest import begin_date_str_with_time
+from .conftest import end_date_str
+from .conftest import end_date_str_with_time
+from .conftest import get_test_date
+from code42cli.click_ext.types import MagicDate
+from code42cli.date_helper import round_datetime_to_day_end
+from code42cli.date_helper import round_datetime_to_day_start
+
+one_ms = timedelta(milliseconds=1)
+
+
+class TestMagicDateNoRounding:
+    md = MagicDate()
+
+    def convert(self, val):
+        return self.md.convert(val, ctx=None, param=None)
+
+    def test_when_given_date_str_parses_successfully(self):
```

Comment:
glaring lack of negative tests in this whole file

Reply:
if we add case insensitivity, we should have tests for that too.
There should also be at least one test for each rounding strategy that verifies that this fails under conditions where it's supposed to

Reply:
Good call, as adding the negative tests did catch some cases we weren't handling correctly. I've tweaked the regex to be a bit tighter around what we're grabbing, and also made it so we accept `2020-01-01T12:00:00` as a valid format.